articles/ros2_threat_model.md

```diff
@@ -0,0 +1,3291 @@
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">3</td>
+    <td class="warning">2</td>
+    <td class="success">1</td>
+    <td class="danger">3</td>
+    <td class="danger">3</td>
+    <td>12</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td>
+      <ul>
+        <li>Actuators should embed an identifier.</li>
+        <li>Command vector should be signed (ideally encryped) to prevent
+manipulation.</li>
+      </ul>
+    </td>
+    <td> </td>
+  </tr>
+
+  <tr>
+    <td>An attacker intercepts the robot actuators command. (can recompute
+localization)</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">1</td>
+    <td class="warning">2</td>
+    <td class="success">1</td>
+    <td class="danger">3</td>
+    <td class="danger">3</td>
+    <td>10</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td>
+      <ul>
+        <li>Command vector should be encrypted.</li>
+      </ul>
+    </td>
+    <td> </td>
+  </tr>
+
+  <tr>
+    <td>An attacker sends malicious command to actuators to trigger the
+E-Stop</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="warning">2</td>
+    <td class="warning">2</td>
+    <td class="danger">3</td>
+    <td class="danger">3</td>
+    <td class="success">1</td>
+    <td>11</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td>
+      <ul>
+        <li>If a joint command is exceeding the joint limits, a specific code
```

Comment:
I don't quite see how this could be realized @thomas-moulard. Do you imply that all trajectories should be monitored for an "out of bounds" potential action and in case such is found a particular routine should get executed? Which routine? Wouldn't it be simpler to just discard the trajectory completely?

Reply:
@vmayoral, line 2717?

Reply:
Maybe I wasn't clear enough, sorry, I don't quite understand how to implement the countermeasure proposed and I'm arguing about whether simply discarding the trajectory would be safer. Does that make more sense @olaldiko?

Reply:
That has safety implications. Simply discarding the trajectory may not be safe in all conditions or for all systems.

Monitoring every single command input (e.g. every trajectory) for dangerous ones is a common technique in safety.

Reply:
Yes, @gbiggs is correct. Coming from a humanoid robotic background, stopping a walking robot can be more dangerous than allowing a slight violation of a joint max velocity or effort. The same idea can apply to any robot where "velocity = 0" is not a safe state (drones, planes, self-driving cars on a highway).

Reply:
All right, that makes sense, however does the same rationale apply for this mobile robot application? It feels like in this particular case we may want to elaborate? I still believe I’d rather have the TB3 stopping than falling/hitting something.